Packages cleanup, convert boot script to systemd service, pass a few OVAL definitions.

cwjenkins opened the pull request:
1) Several packages were not being installed even though they were defined within the packages segment.
    Moved those packages to post install segment to be manually installed.
2) Several packages were not found (ccid, unix2dos, dos2unix). Removed.
3) Some packages were stating they were not found in the package segment, but were installed anyhow (e.g. aide).
4) Converted boot script that disables wifi to a systemd service to avoid tampering with permissions on `/etc/rc.local` (causes oval def oval:ssg-rpm_verify_permissions:def:1 to fail)
5) Added vsftpd, ntp to pass ovals [oval:ssg-package_vsftpd_installed:def:1, oval:ssg-package_ntp_installed:def:1]
6) Removed openssh-server, kexec-tools to pass ovals [oval:ssg-package_openssh-server_removed:def:1, oval:ssg-package_kexec-tools_removed:def:1]

fcaviggia commented:
Overcome by events.

cwjenkins commented:
@fcaviggia not that I'm opposed to closing, but could you elaborate or direct me to something that explains 'overcome by events' ?

fcaviggia commented:
The changes would conflict with the methodology of installing packages at install time - furthermore, the packages removed are required for most DoD/IC shops (specifically IPsec/libreswan). Disabled radios are also a requirement for security (unless you're using a home server on wireless).

cwjenkins commented:
Thanks @fcaviggia, one more question if you could, why would those packages fail to install for me? Is it due to the fact they do not exist in the install ISO I selected (centos7_minimal_1511). That is why I moved them into manual `yum installs` in the post install section.
As for the disabled radio I moved it over to a systemd service which does disable it.
In any event, thanks for your work on this! Saved me a ton of time.

fcaviggia commented:
Yeah, it was designed to be embedded on the DVD - look at hardened-centos.cfg, line 125 copies the RPMs over from hardening directory to root, around line 172 it installs the specific RPMs. I am working on a version at MITRE that uses the DVD and packer to create VM and AMI images.

fcaviggia commented:
The above procedure is for any RPMs not included with the DVD installation that can be called at the beginning of the kickstart.

cwjenkins commented:
Gotcha, and nice work with the use of packer. For our particular setup we will need an ISO.
Thanks again!

openface commented:
@fcaviggia I'd love to learn more about how you're using Packer to build hardened images.   Are you planning to release any of this work by chance?

fcaviggia commented:
I have to get that approved by MITRE, as the work was preformed by a specific program to extend that functionality, but the plan would be to open source the kickstarts and packer content. Part of the issue is the program is undergoing FISMA certification for Amazon's government platform
